src/segment-loader.js

```diff
@@ -1139,20 +1139,34 @@ export default class SegmentLoader extends videojs.EventTarget {
     }
   }
 
-  handleCaptions_(simpleSegment, captions, captionStreams) {
-    // Don't need to check for abort since captions are only handled for non partial
-    // appends at the moment (therefore, they will only trigger once a segment is finished
-    // being transmuxed).
-    createCaptionsTrackIfNotExists(
-      this.inbandTextTracks_, this.hls_.tech_, captionStreams);
-    addCaptionData({
-      captionArray: captions,
-      inbandTextTracks: this.inbandTextTracks_,
-      // full segments appends already offset times in the transmuxer
-      timestampOffset: 0
+  handleCaptions_(simpleSegment, captions) {
+    if (this.checkForAbort_(simpleSegment.requestId) ||
+      this.abortRequestEarly_(simpleSegment.stats) ||
+      !captions || captions.length === 0) {
+      return;
+    }
+
+    const segmentInfo = this.pendingSegment_;
+
+    // TODO what happens in this case?
```

Comment:
Would this be a problem at all? Offhand I'd imagine it should be OK to only append captions and not audio/video.